### PR TITLE
Quests active development PR

### PR DESCRIPTION
**Pull request by Krakenied**
Closes https://github.com/LMBishop/Quests/issues/613 (added new options)
Closes https://github.com/LMBishop/Quests/issues/683 (added support for `-` in names)
Closes https://github.com/LMBishop/Quests/issues/686 (added dedicated version getter)
Closes https://github.com/LMBishop/Quests/issues/684 (added dedicated version getter)
Closes https://github.com/LMBishop/Quests/issues/681 (added dedicated version getter)
Closes https://github.com/LMBishop/Quests/issues/697 (added dedicated version getter)
Closes https://github.com/LMBishop/Quests/issues/698 (added dedicated version getter)
Closes https://github.com/LMBishop/Quests/issues/570 (fixed)
Closes https://github.com/LMBishop/Quests/issues/565 (added `trading` task type)
Closes https://github.com/LMBishop/Quests/issues/542 (added PBT & CoreProtect to farming)
Closes https://github.com/LMBishop/Quests/issues/680 (drop items if cant fit into inventory)
Closes https://github.com/LMBishop/Quests/issues/708 (added `vaultreward` option)
Closes https://github.com/LMBishop/Quests/issues/415 (added even more options to `interact`)

### CODE REVIEW

File: bukkit/libs/PyroFishingPro-4.9.7.jar
Comment (Krakenied) [inline bukkit/libs/PyroFishingPro-4.9.7.jar:1]: Yeah, it's API-only JAR.

**Comment by Krakenied** [comment]
Auxilor repo seems to be down

**Comment by LMBishop** [comment]
@Krakenied, given as you're quite keen in maintaining Quests, would you like to be added as a collaborator to this repository so you can have write permissions? I think it's quite clear that my own inactivity is actively harming the project, and I don't see myself becoming less busy anytime soon.

**Comment by Krakenied** [comment]
Yes, I'd be happy to join as a collaborator to the repository. It would definitely make maintaining the project a bit easier and more convenient. However keep in mind that new versions (releases) still need to be released as most users download the plugin from Spigot (and don't even know about actions tab existence).

**Comment by LMBishop** [comment]
> Yes, I'd be happy to join as a collaborator to the repository. It would definitely make maintaining the project a bit easier and more convenient. However keep in mind that new versions (releases) still need to be released as most users download the plugin from Spigot (and don't even know about actions tab existence) which is literally the main cause of harming the project.

Okay cool, I've sent an invitation to join - you should be able to push branches, update issues, review/merge PRs etc. (let me know if you can't)! You should also be able to create GitHub releases yourself, please feel free to bump the version number whenever you see fit. 

I'm happy to continue publishing the updates on Spigot and all the other places people get Quests from, just let me know whenever you want me to do so. It's probably easiest to just ping me on Discord.

**Review by LMBishop — APPROVED** [review]

**Comment by LMBishop** [comment]
> I definitely can, however you need to submit an approving review for this PR, as I can't review my own PRs.

I've updated the protection rules so this shouldn't be a problem anymore

**Pull request merged**